File: javascript/core/src/engine/verbs/unroll.ts
/*!
 * Copyright (c) Microsoft. All rights reserved.
 * Licensed under the MIT license. See LICENSE file in the project.
 */
import type { UnrollArgs } from '../../types.js'
import { makeStepNode } from '../factories.js'

export const unroll = makeStepNode<UnrollArgs>((input, { columns }) =>
	input.unroll(columns),
)

File: python/data-wrangling-components/data_wrangling_components/engine/verbs/unroll.py
#
# Copyright (c) Microsoft. All rights reserved.
# Licensed under the MIT license. See LICENSE file in the project.
#

from typing import List

from data_wrangling_components.engine.verbs.verb_input import VerbInput
from data_wrangling_components.table_store import TableContainer


def unroll(input: VerbInput, columns: List[str]):
    input_table = input.get_input()
    output = input_table.explode(columns)
    return TableContainer(table=output)
